codec-http2/src/main/java/io/netty/handler/codec/http2/Http2ConnectionHandler.java

```diff
@@ -210,8 +188,13 @@ public final T build(Http2Connection connection) {
                 writer = new Http2OutboundFrameLogger(writer, frameLogger);
             }
             Http2ConnectionEncoder encoder = new DefaultHttp2ConnectionEncoder(connection, writer);
-            if (encoderEnforceMaxConcurrentStreams()) {
-                encoder = new StreamBufferingEncoder(encoder, encoderMaxConcurrentStreams);
+            if (encoderEnforceMaxConcurrentStreams) {
+                if (server) {
```

Comment:
`server` -> `connection.isServer()`.

The builder is a bit strange in that the `server` boolean is only used to construct the `Http2Connection`. If a `Http2Connection` is provided as an argument to the build method we should use the `isServer()` method to determine if what is built will run in server mode or not.